src/main/java/inf112/skeleton/app/model/GameModel.java

```diff
@@ -1,30 +1,43 @@
 package inf112.skeleton.app.model;
 
+import com.badlogic.gdx.utils.Timer;
 import inf112.skeleton.app.Constants;
 import inf112.skeleton.app.model.board.Direction;
 import inf112.skeleton.app.model.board.Location;
 import inf112.skeleton.app.model.board.MapHandler;
+import inf112.skeleton.app.model.board.MoveInstruction;
 import inf112.skeleton.app.model.cards.IProgramCard;
 import inf112.skeleton.app.model.cards.MoveForwardCard;
 import inf112.skeleton.app.model.tiles.TileType;
 
+import java.util.ArrayList;
 import java.util.Deque;
 import java.util.LinkedList;
 import java.util.List;
 
 public class GameModel {
 
-    private final Robot robot; // todo: soon: switch to a list of robots
-    private final MapHandler tiledMapHandler;
-    private final Player player;
-    private List<Robot> robots;
+    private LinkedList<Robot> robots;
+    private Robot robot;
+    private MapHandler tiledMapHandler;
+    private Player player;
+
+    private ArrayList<Deque<MoveInstruction>> cardSteps = new ArrayList<>();
+    private ArrayList<Deque<MoveInstruction>> tileSteps = new ArrayList<>();
+
+
 
     public GameModel(String map_filename) {
         robots = new LinkedList<>();
         robot = new Robot();
         robots.add(robot);
         player = new Player();
         player.generateCardHand();
+        tiledMapHandler = new MapHandler("map-1.tmx");
+        for (int i = 0; i < 5; i++) {
```

Comment:
The number of phases could be a field variable since it's used a couple of times.
```java
private int NUMBER_OF_PHASES = 5
```